libcoz/profiler.h

```diff
@@ -223,6 +227,7 @@ class profiler {
   spinlock _latency_points_lock;  //< Spinlock that protects the latency points map
 
   static_map<pid_t, thread_state> _thread_states;   //< Map from thread IDs to thread-local state
+  uint64_t _num_threads_running;                    //< Number of threads that are currently being sampled
```

Comment:
This should be atomic since it's shared across threads. `uint64_t` is also inconsistent with the other integer sizes used here.

```suggestion
  std::atomic<size_t> _num_threads_running;                    //< Number of threads that are currently being sampled
```